src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -18,273 +18,375 @@
+    ctx.mCommandHandler.AddResponseData(ctx.mRequestPath, response);
+}
 
-            VerifyOrExit(credentials.size() <= sizeof(sNetworks[i].mData.mWiFi.mCredentials),
-                         err = NetworkCommissioningStatus::kOutOfRange);
-            memcpy(sNetworks[i].mData.mWiFi.mCredentials, credentials.data(), credentials.size());
+void Instance::HandleConnectNetwork(HandlerContext & ctx, const Commands::ConnectNetwork::DecodableType & req)
+{
+    Commands::ConnectNetworkResponse::Type response;
 
-            using WiFiCredentialsLenType = decltype(sNetworks[i].mData.mWiFi.mCredentialsLen);
-            VerifyOrExit(CanCastTo<WiFiCredentialsLenType>(ssid.size()), err = NetworkCommissioningStatus::kOutOfRange);
-            sNetworks[i].mData.mWiFi.mCredentialsLen = static_cast<WiFiCredentialsLenType>(credentials.size());
+    mAsyncCommandHandle = app::CommandHandler::Handle(&ctx.mCommandHandler);
+    mpWirelessDriver->ConnectNetwork(req.networkID, this);
+}
 
-            VerifyOrExit(ssid.size() <= sizeof(sNetworks[i].mNetworkID), err = NetworkCommissioningStatus::kOutOfRange);
-            memcpy(sNetworks[i].mNetworkID, sNetworks[i].mData.mWiFi.mSSID, ssid.size());
+void Instance::HandleReorderNetwork(HandlerContext & ctx, const Commands::ReorderNetwork::DecodableType & req)
+{
+    Commands::NetworkConfigResponse::Type response;
+    response.networkingStatus = ToClusterObjectEnum(mpWirelessDriver->ReorderNetwork(req.networkID, req.networkIndex));
+    ctx.mCommandHandler.AddResponseData(ctx.mRequestPath, response);
+}
+
+void Instance::OnResult(Status commissioningError, CharSpan errorText, int32_t interfaceStatus)
+{
+    auto commandHandleRef = std::move(mAsyncCommandHandle);
+    auto commandHandle    = commandHandleRef.Get();
+    if (commandHandle == nullptr)
+    {
+        // Oh, this is a response of a network scan from unclean shutdown, ignore it.
+        return;
+    }
 
-            using NetworkIDLenType = decltype(sNetworks[i].mNetworkIDLen);
-            VerifyOrExit(CanCastTo<NetworkIDLenType>(ssid.size()), err = NetworkCommissioningStatus::kOutOfRange);
-            sNetworks[i].mNetworkIDLen = static_cast<NetworkIDLenType>(ssid.size());
+    Commands::ConnectNetworkResponse::Type response;
+    response.networkingStatus = ToClusterObjectEnum(commissioningError);
+    response.debugText        = errorText;
+    response.errorValue       = interfaceStatus;
+    commandHandle->AddResponseData(mPath, response);
+}
 
-            sNetworks[i].mNetworkType = NetworkType::kWiFi;
-            sNetworks[i].mEnabled     = false;
+void Instance::OnFinished(Status status, CharSpan debugText, ThreadScanResponseIterator * networks)
+{
+    CHIP_ERROR err        = CHIP_NO_ERROR;
+    auto commandHandleRef = std::move(mAsyncCommandHandle);
+    auto commandHandle    = commandHandleRef.Get();
+    if (commandHandle == nullptr)
+    {
+        // Oh, this is a response of a network scan from unclean shutdown, ignore it.
```

Comment:
Can the exact details of object lifecycle be explained here so it's clear what's going on?  This comment just basically says that this is an exceptional case that can be ignored and won't hurt anything.  But that doesn't clearly explain how things are working and why it is OK to ignore this.

Reply:
Updated the comment to explain why it is necessary to ignore this.